### ISSUE TITLE
v2 couple of Bugs & Feature Requests

### ISSUE THREAD

**Issue by hinneLinks** [chubin/wttr.in#440]
I like v2 a lot, thanks for it!
I have some feature requests, also i found some bugs.

**Bugs:**

1. The current time in the .png view contains a Placeholder `%{{NOW(Europe/Berlin)}}` instead of the time, see attachment
![wttr in-Frankfurt,Main](https://user-images.githubusercontent.com/15357388/81084405-17c35e00-8ef6-11ea-86f0-57ccafdd5d21.png)

2. On Ubuntu 18.04, in the default terminal, some Icons (current weather and Wind direction) are too close to the Text, see screenshot:
![wttr in-terminal-ubuntu-18 04](https://user-images.githubusercontent.com/15357388/81084868-bfd92700-8ef6-11ea-9e3a-282ffab47fc9.png)

3. The plotted diagram seems to always start with a vertical line?
4. Sometimes, there are outbreaks, look at the pictures at 05 May, 6 o'clock

**Feature Requests:**
1. Have a Wiki or some help, that explains the output. So far I've found these Explanations:
    a) [Blue shades](https://github.com/chubin/wttr.in/issues/385)
    b) [White lines at the bottom](https://github.com/chubin/wttr.in/issues/373)
    c) the row with Numbers seems to be the wind speed, the row before that the wind direction?
2. In the plotted diagram its hard to guess what temperature it will be, lets say this afternoon. Could we have some Numbers in the Diagram? Or maybe add more numbers tho the axis on the left? I've made a Mock-Up for the first one.
![image](https://user-images.githubusercontent.com/15357388/81086678-1e9fa000-8ef9-11ea-8f26-382ba54133c8.png)

**Comment by chubin** [comment]
Thank you for your comments, and for the interest to our gemütlichen service :)
First of all, it is not yet finished, it is just a beta, or pre-beta, that we are playing with;
it was published just to gather some feed back and criticism. So thank you for it.

1. The problem with the placeholder is known; it will be fixed soon.
The problem here, is that because of the high load that the service has,
it was partially rewritten from Python, in Go, and the PNG rendering part is still in Python.
The go service is acting like a HTTP/HTTPS-frontend + it makes some caching
where possible. Before PNG-rendering will be rewritten in Go, the best solution probably
could be to switch the caching of the v2 PNGs off (whould should have no impact on the service).

2. Yes, I believe, spaces here would be a good solution (the problem is related to #433; @hern42 what do you think about it?)

3. It is a bug. It must be fixed.

4. It seems to be a bug too.

FR1. Yes, I agree, I think this information should be added to /:help and to the README file.

FR2. Not sure about this one. My idea was to mark the max and min values for each day,
like it is done for rate.sx (check `curl rate.sx/btc` to see how it looks like).

Other visual features, that are still not implemented (JFYI):

1. Sun trajectory is not yet implemented.
2. The azure vertical line indicating the current time must be in full height.

**Comment by hern42** [comment]
hi.
ah! some more spacing issues... nice...

i did a bit of searching and i couldn't find any "canonical way" of doing this. the consensus seems to be to consider an emoticon as a word and therefore put a space before and after when put amongst other words. however that gets tricky considering punctuation. what about the period or the coma! again, no clear rule. if the emoticon is rendered there is not much of an issue (except ugliness) but if it, for whatever reason, stays as a combination of alphanumerical characters that is asking for trouble... ;). <---- my point.

in this particular (2.) problem i'd suggest indeed a space for the main weather emoticon. however, i'd keep the arrow stuck to the wind speed as for me it is basically similar to - (minus) or + (plus) in front of a temperature. sort of.

that was my 2 cents.

i'd suggest maybe as a(n unofficial) feature request to add an option for "compacity" for those who are in trouble for displaying this piece of data somewhere tight (as, in my case, a fairly narrow conky on the side of my screen, using the format option)

hope it helps...
cheers,
a-

**Comment by hinneLinks** [comment]
Bug2: Mainly it seems, that my (default) font is the problem, as it allows two characters to merge into each other (the arrow seems to be inside of the 2).
Maybe some Zalgo can help? https://en.wikipedia.org/wiki/Combining_character

I've tried some half-width-spaces in my terminal, but all those have the same length (which makes sense, since its all mono spaced)

FR2:
Yeah, at least some indication what the min/max each day is would be nice. Or You could have a legend for the colors ,e,g Blue=1-4 Green=5-8, yellow9-11, orange 11-14, red 15-18

**Comment by chubin** [comment]
Zalgo will not work in your terminal, because your terminal is VTE-based (it is gnome-terminal, right?), and it does not support combining characters. Only Konsole (from the popular terminals) does.
It is interesting indeed why you have this issue with the wind direction. It is displayed correctly in all my terminals, and PNG files do not seem to have this issue either.

The rest is accepted. Min/Max temperature should be indicated; a space after the weather symbol should be added as @hern42 suggests

**Comment by hinneLinks** [comment]
Yes, its gnome terminal 3.28.2. Zlago does Work, see Screenshot (the vertical crazy signs only appear after a focus change).
![image](https://user-images.githubusercontent.com/15357388/81191352-39d1e480-8fb9-11ea-986a-63a24b6b46f6.png)

Its the Default-Font, Ubuntu Mono Regular i assume, that causes the Issue. If I use one of the other Fonts (like Monospace Regular) the wind direction has more space:
![image](https://user-images.githubusercontent.com/15357388/81191653-8d443280-8fb9-11ea-8c0c-cd99c7fbb134.png)

So Bug 2 can be ignored as user-problem i guess.

**Comment by chubin** [comment]
Interesting. Can you please check if `curl hi.wttr.in/?Q0` works for you too?

gnome-terminal (wrong):

![Selection_040](https://user-images.githubusercontent.com/3875145/81193734-1492a580-8fbc-11ea-8959-237034aab70e.png)

konsole (correct):

![Selection_041](https://user-images.githubusercontent.com/3875145/81193860-3a1faf00-8fbc-11ea-9281-f30de4de53da.png)

**Comment by hinneLinks** [comment]
Not sure whats wrong with this, but it looks like your gnome-terminal. Using a different font dosn't change the location of the gibberish like konsole does.
![image](https://user-images.githubusercontent.com/15357388/81207830-7cea8280-8fce-11ea-89c3-a750fa8dbf13.png)

**Comment by chubin** [comment]
You have this problem too, you see? These two circles should not be displayed; that's because gnome-terminal (your version) does not support it.

BTW, another good terminal emulator that supports it (besides Konsole)
is alacritty:

![Screenshot_2020-05-07_22-11-29](https://user-images.githubusercontent.com/3875145/81340278-db3b6200-90af-11ea-98c8-66af0ae92751.png)

The alignment should be fixed, but Hindi support is anyway not yet fully
implemented, because of the mentioned problem

**Comment by chubin** [comment]
(Added "todo" to the issue description)

**Comment by pabloab** [comment]
Maybe this recent (trivial?) bug on v2 is related to this issue:

![image](https://user-images.githubusercontent.com/657836/82185339-71953200-98bf-11ea-967b-f8414101322e.png)

**Comment by chubin** [comment]
@pabloab This bug is not so trivial as it may seem, because output depends on the terminal. Some symbols have different width in different terminals, and it is hard to say on the server side, how exactly the output should be rendered. 

To see it yourself, you can just query different cities (so that different emojis will be used),
and you will see that this misalignment differs from location to location.

It is possible to find a solution, but it will be not trivial

**Comment by mckaymatt** [comment]
Could v2 get support for USCS units? 
Also thanks for making this. V2 is great and I already find it really useful, despite the fact that I'm far more comfortable with Fahrenheit.

**Comment by chubin** [comment]
@mckaymatt I will add it, but you are right; currently USCS is not supported
